js/components/upNextPanel.js

```diff
@@ -8,127 +8,97 @@
 * @constructor
 */
 var React = require('react'),
+    ClassNames = require('classnames'),
     CONSTANTS = require('./../constants/constants'),
     Utils = require('./utils'),
-    InlineStyle = require('../styles/inlineStyle'),
+    CloseButton = require('./closeButton'),
     CountDownClock = require('./countDownClock');
 
 var UpNextPanel = React.createClass({
-  getInitialState: function() {
-    this.isMobile = this.props.controller.state.isMobile;
-    return {
-      contentDescription: this.props.upNextInfo.upNextData.description
-    };
+  propTypes: {
+    upNextInfo: React.PropTypes.shape({
+      upNextData: React.PropTypes.shape({
+        preview_image_url: React.PropTypes.string,
+        name: React.PropTypes.string,
+        description:React.PropTypes.string
+      })
+    }),
+    skinConfig: React.PropTypes.shape({
+      upNext: React.PropTypes.shape({
+        timeToShow: React.PropTypes.number
+      }),
+      icons: React.PropTypes.objectOf(React.PropTypes.object)
+    })
   },
 
-  componentDidMount: function() {
-    var descriptionNode = this.refs.ContentDescription.getDOMNode();
-    var shortDesc = Utils.truncateTextToWidth(descriptionNode, this.state.contentDescription);
-    this.setState({contentDescription: shortDesc});
-
-    if (Utils.isSafari()){
-     InlineStyle.upNextPanelStyle.upNextTitleStyle.display = "-webkit-flex";
-    }
-    else {
-     InlineStyle.upNextPanelStyle.upNextTitleStyle.display = "flex";
-    }
+  getDefaultProps: function () {
+    return {
+      skinConfig: {
+        upNext: {
+          timeToShow: 10
+        },
+        icons: {
+          play:{fontStyleClass:'icon icon-play'},
+          dismiss:{fontStyleClass:'icon icon-close'}
+        }
+      },
+      upNextInfo: {
+        upNextData: {}
+      },
+      controller: {
+        upNextDismissButtonClicked: function(){},
+        sendDiscoveryClickEvent: function(a,b){}
+      }
+    };
   },
 
-  closeUpNextPanel: function(event) {
-    if (event.type == 'touchend' || !this.isMobile){
-      //since mobile would fire both click and touched events,
-      //we need to make sure only one actually does the work
-
-      OO.log("Up next panel close button clicked");
-      this.props.controller.upNextDismissButtonClicked();
-    }
+  closeUpNextPanel: function() {
+    this.props.controller.upNextDismissButtonClicked();
```

Comment:
that's correct, de-duping no longer needed because onClick event is attached to a clickable element `<button>` instead of `<div>` or `<span>`